### PR TITLE
feat(components): [el-image] support append preview to body

### LINKED ISSUE #5879: [Bug Report]  图片预览大图遮罩层和表格的border css层级冲突
### Element Plus version
2.0.1

### OS/Browsers version
Windows 10 / Chrome 98.0.4758.82（正式版本） （64 位）

### Vue version
3.2.27

### Reproduction Link
https://codepen.io/zickCode/pen/WNXpXPo

### Steps to reproduce
1.table组件中，引入image组件 ，配置查看大图属性preview-src-list。
2.点击图片，弹窗的遮罩层未能完全挡住table组件的边框样式

### What is Expected?
遮罩层应该层级更高

### What is actually happening?
样式冲突

### PR DESCRIPTION
**Pull request by msidolphin**
fix #5879

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/image/src/image.vue [hunk 1]
```diff
@@ -61,6 +67,17 @@ export default defineComponent({
...
+    useDeprecated(
+      {
+        scope: 'el-image',
+        from: 'append-to-body',
+        replacement: 'preview-append-to-body',
+        version: '2.2.0',
+        ref: 'https://element-plus.org/en-US/component/image.html#image-attributess',
+      },
+      props.appendToBody
+    )
```
Comment (JeremyWuuuuu) [inline packages/components/image/src/image.vue:79]: After warning about the deprecation, we should use a compatible version for this variable:

```ts
const compatibleAppendToBody = computed(() => {
  isBoolean(props.appendToBody) ? props.appendToBody : props.previewAppendToBody
})

```

**Review by jw-foss — COMMENTED** [review]
Left a comment for you ;)

**Comment by msidolphin** [comment]
@JeremyWuuuuu done

File: packages/components/image-viewer/src/image-viewer.ts [hunk 2]
```diff
@@ -22,6 +21,10 @@ export const imageViewerProps = buildProps({
     type: Boolean,
     default: false,
   },
+  appendToBody: {
+    type: Boolean,
+    default: false,
```
[the remark this answers is not in the document]
Comment (msidolphin) [reply packages/components/image-viewer/src/image-viewer.ts:26]: > I think we should set the default value to `undefined` which allows the deprecation warning to throw, without setting it `undefined` the deprecation warning will always be thrown which could be annoying for the developers.

teleported is new property, is it necessary?

**Review by jw-foss — APPROVED** [review]
I am approving this PR, when the CI gets passed, please merge it yourself.

File: packages/components/image-viewer/src/image-viewer.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (JeremyWuuuuu) [reply packages/components/image-viewer/src/image-viewer.ts:26]: If you don't do this, how could you tell if `append-to-body` is used and warn the user about the deprecation then?

**Comment by JeremyWuuuuu** [comment]
@msidolphin Please rebase the dev again for getting this https://github.com/element-plus/element-plus/pull/5960 patch

File: packages/components/image-viewer/src/image-viewer.vue [hunk 3]
```diff
@@ -89,12 +91,11 @@ import {
   effectScope,
   markRaw,
 } from 'vue'
-import { useEventListener } from '@vueuse/core'
-import { throttle } from 'lodash-unified'
```
Comment (JeremyWuuuuu) [inline packages/components/image-viewer/src/image-viewer.vue:93]: @msidolphin Seems like you accidentally removed `throttle` importing.

**Review by jw-foss — APPROVED** [review]

**Pull request merged**